assets/semgrep_rules/client/brave-execute-script.yaml

```diff
@@ -0,0 +1,21 @@
+rules:
+  - id: brave-execute-script
+    metadata:
+      author: Andrea Brancaleoni <abc@pompel.me>
+      references:
+        - https://github.com/brave/brave-browser/wiki/Security-reviews
+      confidence: MEDIUM
+      source: https://github.com/brave/security-action/blob/main/assets/semgrep_rules/client/brave-execute-script.yaml
+    message: |
+      $FUNC usages should be vet by the security-team.
+
+      References:
+      - https://github.com/brave/brave-browser/wiki/Security-reviews (point 13)
+    severity: INFO
+    languages:
+      - cpp
+    patterns:
+      - pattern: $OBJ.$FUNC(...)
+      - metavariable-regex:
+          metavariable: $FUNC
+          regex: ^ExecuteScript|ExecuteScriptInIsolatedWorld$
```

Comment:
@bridiver anything else to add here?

Reply:
`RequestExecuteScript` on the renderer side. I don't see any code in brave or chromium that includes the string `ExecuteScript` that does not in fact execute JS so I think we can use that by itself with no ^ or $

Reply:
using `ExecuteScript` will also help catch "helper" methods that pass scripts along to actual instance of ExecuteScriptInIsolatedWorld, etc...
There's also `ExecuteMethodAndReturnValue` and `CallFunctionEvenIfScriptDisabled` which execute v8::Functions instead of raw strings

Reply:
although for those cases we might want to look for `v8::Function` instead of `ExecuteMethodAndReturnValue`/`CallFunctionEvenIfScriptDisabled` and we can see if that generates too many false positives or not

Reply:
Will go for the method names for the moment. We can have a separated rule for `v8::Function` which will be noisy for sure